ch_pipeline/processing/daily.py

```diff
@@ -406,21 +408,110 @@ def _load_hook(self):
         # Process the intervals
         self._intervals = []
         for t in self._revparams["intervals"]:
-            self._intervals.append((t["start"], t.get("end", None)))
+            self._intervals.append((t["start"], t.get("end", None), t.get("step", 1)))
+
         self._padding = self._revparams["padding"]
 
+    def _available_files(self, start_csd, end_csd):
+        """
+        Return chimestack files available in cedar_online between start_csd and end_csd, if all of the files for that period are available online.
+
+        Return an empty list if files between start_csd and end_csd are only partially available online.
+
+        Total file count is verified by checking files that exist everywhere.
+
+        Parameters
+        ----------
+        start_csd : int
+            Start date in sidereal day format
+        end_csd : int
+            End date in sidereal day format
+
+        Returns
+        -------
+        list
+            List contains the chimestack files available in the timespan, if all of them are available online
+
+        """
+
+        # Connect to databases
+        db.connect()
+
+        # Get timestamps in unix format
+        # Needed for queries
+        start_time = ephemeris.csd_to_unix(start_csd)
+        end_time = ephemeris.csd_to_unix(end_csd)
+
+        # We will want to know which files are in chime_online and nearline on cedar
+        online_node = [di.StorageNode.get(name="cedar_online", active=True)]
+
+        # TODO if the time range is so small that it’s completely contained within a single file, nothing will be returned
+        # have to special-case it by looking for files which start before the start time and end after the end time).
+
+        archive_files = (
+            di.ArchiveFileCopy.select(
+                di.ArchiveFileCopy.node,
+                di.ArchiveFile,
+                di.ArchiveAcq,
+                di.ArchiveInst,
+                di.CorrFileInfo,
+            )
+            .join(di.ArchiveFile)
+            .join(di.ArchiveAcq)
+            .join(di.ArchiveInst)
```

Comment:
This join seems pointless and there's enough table merging going on already.  Yes, you can isolate chimestack files by looking at the instrument name, but that's also encoded in the acq name, which you've already included in your query.

I'd drop the `ArchiveInst` join completely and then change your `where()` criterion below from
```
di.ArchiveInst.name == "chimestack"
```
to
```
di.ArchiveAcq.name % "%_chimestack"
```
which will match all acquisitions whose name ends in "chimestack".

Reply:
If you _really_ don't want to rely on the acq name to select chimestack, then it'd be better to handle this like you do with the node selection.  Fetch the `chimestack` instrument record once:
```
chimestack_inst = di.ArchiveInst.get(name="chimestack")
```
and then use that in the where clause, again to avoid this unnecessary join:
```
where(
   # [...]
   di.ArchiveAcq.inst == chimestack_inst
   # [...]
   )
```
You might even be able to argue this is more efficient than pattern matching against acq name like I suggested above.

Reply:
Yeah, the second is better.  Mariadb's pattern matching is probably pretty good, but it's still going to be significantly heavier than an integer comparison.